src/core/common/logging.hpp

```diff
@@ -110,6 +110,21 @@ extern "C" {
 #define _OT_REGION_BBR_PREFIX _OT_REGION_SUFFIX
 #endif
 
+/**
+ * @def otLogNone
+ *
+ * Logging at log level none.
+ *
+ * @param[in]  aRegion   The log region.
+ * @param[in]  ...       Arguments for the format specification.
+ *
+ */
+#if OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_NONE
+#define otLogNone(aRegion, ...) _otLogFormatter(OT_LOG_LEVEL_NONE, aRegion, _OT_LEVEL_NONE_PREFIX __VA_ARGS__, NULL)
+#else
+#define otLogNone(aRegion, ...)
+#endif
```

Comment:
Log level "none" indicates that user does not want any logging (i.e., disable all logging). So having an `otLogNone()` and `otLogNoneCore()` macros to emit logs at "none" level does seem good to me.

Also note that by using `otLog{Level}Core` you are tied to `OPENTHREAD_CONFIG_LOG_CORE` config and if that is disabled, then OTNS logs wont be emitted as well.

I guess you want to emit logs for OTNS independent of log level and all other configs (this is similar to cert log I'd say).
So I would suggest adding  a specific macro for OTNS, something like this:
```
#if (OPENTHREAD_MTD || OPENTHREAD_FTD) && OPENTHREAD_CONFIG_OTNS_ENABLE
#define otLogOtns(...)   _otLogFormatter(OT_LOG_LEVEL_NONE, OT_LOG_REGION_CORE, _OT_LEVEL_NONE_PREFIX __VA_ARGS__, NULL)
#endif
```
(would be good to add it conditionally when OTNS config is enabled)

Reply:
Hmm, I was under the impression that "none" is the highest level of logging. In `wpantund` it is interpreted as "emergency" logging level. Your solution does make more sense however.

Reply:
Edit: Noticed I had a typo -> "to emit logs at "none" level does seem good to me" -> "to emit logs at "none" level does **_not_** seem good to me". 

"none" is not really a log level to be used within OpenThread itself. It indicates that user wants no logging (disable all logging).

Reply:
Maybe I can settle with critical level then?

Reply:
Discovered that I need to avoid using debug level logging, since the verbose logging exceeds the NCP buffer size and will cause OTNS logs to be lost. However there's no easy CLI build flags to set logging levels yet. Falling back to use none level as previously suggested.

Reply:
- There is a CLI command `"log level"` to get or change log level dynamically. 
- Note that you need `OPENTHREAD_CONFG_LOG_LEVEL_DYNMIAC_ENABLE` enabled to allow dynamic log level change (during run time) 
- Otherwise the log level is determined through the config `OPENTHREAD_CONFIG_LOG_LEVEL`. 
- You can check `config/logging.h` for all config related to logging behavior. 
- Some of these configs are also exposed through cmake options (`OT_LOG_LEVEL_DYNMIAC`, `OT_FULL_LOGS`) and autoconf switches.

Reply:
Do you suggest I use critical level for this log? The only purpose for this is to enable real devices to emit status logs to wpantund where it will be redirected to OTNS by Silk to allow for network visualization when running test cases. Ideally it should be turned on whenever I set `OTNS=1`. A change of `OPENTHREAD_CONFIG_LOG_LEVEL` isn't supported as one of the make command flags, right?
Also `OT_FULL_LOGS` does flood the NCP buffer and cause loss of logs as I mention above.

Reply:
Although I think the proper way is to have a new log region for OTNS, and thus enable an easy control of OTNS logs. But I feel like that's a part of the specification and adding a region will be a much more arbitrated process.

Reply:
As I mentioned before, to me this looks like cert logging (`CONFIG_REFERNCE_DEVICE_ENABLE`). 

So similar approach to that makes sense, which is close to what the latest code in this PR is doing, i.e.,
- adding specific logging macro for this use-case (and we can do whatever level/regsion/etc in macro definition)
- and that the macro is only provided when OTNS is enabled (and not tied to any other configs).

Reply:
I think having a log region specific to OTNS makes the most sense. Can you clarify the concern about adding a log region for OTNS? This is an OpenThread-specific concept and nothing to do with the Thread Specification.

Reply:
As @abtink , I'm fine with the current approach as well (given that it is similar to the existing reference device logs).

Reply:
I am not concerned per se, just wasn't sure that I felt comfortable making the decision of adding a new region unless other maintainers agree with it, since I am very new to the codebase. Now that this is merged, a change could be made in the future to add the region if we do have the need to have finer grained control of OTNS logs, or if we are using OTNS logs in more places. For now I think this is fine, as OTNS logs won't be used beyond reporting these status messages and radio events, as far as I can tell.